code/_onclick/hud/fullscreen.dm

```diff
@@ -243,3 +278,241 @@
+	var/special_tick_dir = p_special_tick_dir /// If it gets a special call, if it ticks up or down in order to prevent advantages
+
+	var/create_new_lighting_timer = p_create_new_lighting_timer /// used to prevent timer dupes, keep this as False unless its supposed to be the first call
+	var/lighting_deactivates = p_lighting_deactivates /// If the lighting deactivates
+
+	var/lighting_stage = clamp((floor((ROUND_TIME + stage_time - special_start_time - startup_delay)/stage_time)), 0, max_stages) /// the current stage of the lighting, ticks up by 1 every stagetime after startup_delay + start_time
+	//uses formula (x + y - w - z)/(y) with x = round_time, y = stage_time, w = special_start_time, and z being startup_delay
+
+	var/time_til_next_lighting_call = max(((lighting_stage * stage_time) + startup_delay + special_start_time - ROUND_TIME), 0.5 SECONDS) /// how long until the next sunstage occurs (minimum of 0.5 seconds)
+
+	if(special_call && lighting_stage != 0) // controls stuff related to special calls, prevents people from getting unfair advantages by getting stages reset, unnecessary for short anims
+		if(lighting_deactivates && ROUND_TIME < (stage_time * max_stages) + special_start_time + startup_delay) //if its finished max stage anim and doesn't deactivate, make special calls animate to full
+			lighting_stage = clamp((lighting_stage + special_tick_dir), 0, max_stages)
+		if(time_til_next_lighting_call < special_stage_time)
+			time_til_next_lighting_call = time_til_next_lighting_call + special_stage_time //delays main anims until the special call anim is done
+
+	if(special_lighting == SPECIAL_LIGHTING_SUNSET)
+		lighting_color = special_lighting_sunset(lighting_stage)
+	if(special_lighting == SPECIAL_LIGHTING_SUNRISE)
+		lighting_color = special_lighting_sunrise(lighting_stage)
+
+	if(lighting_stage == 0) //there aren't any cases you won't want these coming up fast
+		special_stage_time = 0.5 SECONDS
+		time_til_next_lighting_call = startup_delay + special_start_time - ROUND_TIME
+
+	if(create_new_lighting_timer) // if create_new_lighting_timer = TRUE, a new timer gets set
+		special_lighting_active_timer = FALSE
+
+	if(!special_lighting_active_timer)
+		if(lighting_stage < max_stages)
+			addtimer(CALLBACK(src, PROC_REF(special_lighting_animate), special_lighting, stage_time, max_stages, startup_delay, special_start_time, null, special_tick_dir, FALSE, TRUE, lighting_deactivates), time_til_next_lighting_call, TIMER_UNIQUE|TIMER_OVERRIDE|TIMER_DELETE_ME)
+			special_lighting_active_timer = TRUE
+		if(lighting_stage == max_stages && lighting_deactivates) // deactives special lighting when the sun hits #000
+			addtimer(CALLBACK(src, PROC_REF(special_lighting_unregister_signals)), time_til_next_lighting_call, TIMER_UNIQUE|TIMER_OVERRIDE|TIMER_DELETE_ME)
+
+	if(CEILING_IS_PROTECTED(lighting_mob_area?.ceiling, CEILING_PROTECTION_TIER_2)) //if underground, don't animate, this is needed in combo with the special area check
```

Comment:
```suggestion
	if(CEILING_IS_PROTECTED(lighting_mob_area?.ceiling, CEILING_PROTECTION_TIER_1)) //if underground, don't animate, this is needed in combo with the special area check
```
Tier 2 protection means that only cave-level areas are in the dark, meaning that all colony buildings are completely bright. changing it to level 1 means that only outside/glassroof gets to see the sun.